src/commands/org/create/sandbox.ts

```diff
@@ -262,17 +293,42 @@ export default class CreateSandbox extends SandboxCommandBase<SandboxCommandResp
         this.flags.wait.seconds,
       ]);
     }
+
+    const sourceIdFlag = !!this.flags['source-id'];
+    const sourceSandboxNameFlag = !!this.flags['source-sandbox-name'];
+
+    const defFileContent = this.flags['definition-file'] ? readSandboxDefFile(this.flags['definition-file']) : {};
+    const sourceIdInDefFile = !!defFileContent.SourceId;
+    const sourceSandboxNameInDefFile = !!defFileContent.SourceSandboxName;
+
+    if ((sourceIdFlag || sourceSandboxNameFlag) && (sourceIdInDefFile || sourceSandboxNameInDefFile)) {
+      throw messages.createError('error.cloneFlagsConflict');
```

Comment:
these validations don't require any specific context other than reading the def.file, let's move these to each source flag, like this:
https://github.com/salesforcecli/plugin-org/blob/0e7d3a7f6ae98240404311fc30a1ed4df3a32eb9/src/commands/org/create/sandbox.ts#L75

you can do more specific error msgs like `Can't use --source-id and define `sourceId` in the def. file` at the same time.